### ISSUE TITLE
Button to manually trigger a cronjob

### ISSUE THREAD

**Issue by taylorsmithgg** [argoproj/argo-cd#7968]
# Summary

It would be extremely useful to have the ability to manually trigger cronjobs after a sync is sucessful.

# Motivation

Our developers are currently pushing cronjobs with argo, but do not have an immediate way to test them and verify. They have to either manually request our assistance to trigger them or wait until the schedule triggers.
Even if we allowed devs to delete/recreate or force, there are no patterns to also create a manual trigger in the process today.

# Proposal

An additional button in the UI would be great. 
We (admins) are currently using k9s to do this.
![image](https://user-images.githubusercontent.com/6597925/146427446-dce04287-7c71-4d9c-8c3e-299168f437be.png)

**Comment by jannfis** [comment]
I first thought this could have been implement as a resource action, such as the ones we already have for [Deployment](https://github.com/argoproj/argo-cd/tree/master/resource_customizations/apps/Deployment/actions) resources. 

But to trigger a Cronjob in K8s requires creating a new `Job` resource, using the `CronJob` as a template, e.g. `kubectl create job --from=cronjob/somecronjob new-job-name`. Resource actions only let you modify the resource they are running for, and do not allow you to create new resources, such as Jobs (for a good reason).

Due to this constraints, I'm not sure this is something we could/should implement in Argo CD. But I'm happy to pick up the discussion.

**Comment by taylorsmithgg** [comment]
Just wanted to bump this for visibility and let you know this is still desperately needed.
Our current strategy is to push just the `Job` into a separate argo manifest.
As you can imagine, it's more tedious and less reliable than having an easy way to just trigger a cronjob.

**Comment by zen** [comment]
Yes, that would be very useful. Lens UI already has this implemented and it's very handy.

**Comment by taylorsmithgg** [comment]
FYI in case anyone else crosses this bridge: argo workflows has a much better management interface and RBAC for jobs. It works out of the box and uses a separate permission set that adheres more closely to the use case.

**Comment by bygui86** [comment]
@taylorsmithgg can you put here a link to that argo workflows feature?
I'm using full argo stack :) so it would be amazing to use what you describe!

**Comment by terrytangyuan** [comment]
https://github.com/argoproj/argo-workflows/blob/master/examples/cron-workflow.yaml

**Comment by bygui86** [comment]
@terrytangyuan thanks.
So Argo Workflow is able to trigger a workflow (aka job) from a cron-workflow (aka cronjob) from the UI for example?

**Comment by terrytangyuan** [comment]
CronWorkflow is not meant to be manually triggered. You can save the workflow spec into a WorkflowTemplate and submit it from the UI.

**Comment by bygui86** [comment]
@terrytangyuan ok got it thanks :)
I don't have the possibility to convert all CronJobs into CronWorkflow, so Argo Workflows is a good alternative, but not replacing the feature request of this thread.

**Comment by jakestonelater** [comment]
An upvote for this feature
Yes, there are workarounds, but the ethos of ArgoCD is to add a convenience wrapper to Kubernetes, right?
I believe this feature would meet that criteria

**Comment by stephaneetje** [comment]
That feature would really be VERY useful
